### ISSUE TITLE
Add a native Edit Connections menu

### ISSUE THREAD

**Issue by tsimonq2** [palinek/nm-tray#23]
Right now, we depend on the GNOME network manager to display the "Edit Connections" menu. This is problematic, because at least on Ubuntu and Debian, this pulls in a fair amount of dependencies.

Could we implement this natively?

**Comment by agaida** [comment]
@tsimonq2 - don't speak for here debian about dependencies - and no, it is not problematic, cmst/connman is the default alternative.

@palinek - but a release would be fine. :)

**Comment by tsimonq2** [comment]
@agaida Are you sure?
```
nm-tray: starting the nm-connection-editor
```

```
$ apt-file find nm-connection-editor
network-manager-gnome: /usr/bin/nm-connection-editor
network-manager-gnome: /usr/share/appdata/nm-connection-editor.appdata.xml
network-manager-gnome: /usr/share/applications/nm-connection-editor.desktop
network-manager-gnome: /usr/share/man/man1/nm-connection-editor.1.gz
```

Yes, by default, it is using that dialogue.

**Comment by agaida** [comment]
I am dead sure that we use connman and cmst in debian since the beginning - so - as i wrote, we have no problem. *You* want nm-tray for Lubuntu and i fully understand it. And i warned you that nm-tray is not prime time ready long ago. One point was the missing native dialogs. 

And it this is fixed i would package and upload nm-tray without problems. That does't mean that i would change the defaults too - but it would be fine for network-manager users to have a pure Qt UI alternative.

**Comment by tsimonq2** [comment]
> And i warned you that nm-tray is not prime time ready long ago. One point was the missing native dialogs.

Uhm... this is me filing the issue for accomplishing that.

Don't shake your finger at me and say "I told you so." I want to make this project better, and I'm sure you do too, but scolding me for testing it isn't a productive way to do so.

Please, let's keep this on-topic.

**Comment by palinek** [comment]
> Yes, by default, it is using that dialogue.

No, you probably overridden the default, see here -> https://github.com/palinek/nm-tray/blob/master/resources/nm-tray.conf#L2

It is dead sure, that we can have our own connections editor, but I think it would take a lot of time/work and I'm lacking free time to implement it...

**Comment by tsimonq2** [comment]
> No, you probably overridden the default

I don't even have that conf file in my packaging. It has not been in a tag yet.

**Comment by palinek** [comment]
> @palinek - but a release would be fine. :)

> I don't even have that conf file in my packaging. It has not been in a tag yet.

..released 0.4.0

**Comment by Myrddin-Wyllt** [comment]
[nmed-qt5.zip](https://github.com/palinek/nm-tray/files/2589842/nmed-qt5.zip)
88e4f6921b935a98879c4fd5f106ddedc3a5005c9c186b42c2cd5b346f2416de4f6466e4c75789646eb2f9860f51c9ee13f74f7c7dd0db6359a79f280daf27ac  nmed-qt5.zip

Here's some rough mockups for the connection editor interfaces. I'm no Qt5/C++ wizard though, I might've gotten includes wrong, and there is no functionality written yet.

**Comment by jubalh** [comment]
@Myrddin-Wyllt Hi there!
Have you worked with GitHub before? I think, it would be nice if you could create a pull request with the mockups, and people could expand on that PR, trying to implement the actual functionalities.

**Comment by palinek** [comment]
>  and there is no functionality written yet. Don't wait on me for the functionality

@Myrddin-Wyllt what is it good for? (to see the possibilities for connection editing we can have a look at `nm-connection-editor`)

**Comment by Myrddin-Wyllt** [comment]
@palinek You're right. My mistake. I've deleted my previous comment. Apologies for the intrusion.

**Comment by cdarken** [comment]
> No, you probably overridden the default, see here -> https://github.com/palinek/nm-tray/blob/master/resources/nm-tray.conf#L2

Where is that config file loaded in the code? Can't find any reference to it. Can I place it in my home dir to be able to modify it?

**Comment by palinek** [comment]
> Where is that config file loaded in the code? Can't find any reference to it.
https://github.com/palinek/nm-tray/blob/master/src/tray.cpp#L235
https://github.com/palinek/nm-tray/blob/master/src/tray.cpp#L277
https://github.com/palinek/nm-tray/blob/master/src/tray.cpp#L351

> Can I place it in my home dir to be able to modify it?
Yes, you can put it into XDG_CONFIG_HOME. See description of the search paths -> https://doc.qt.io/qt-5/qsettings.html#platform-specific-notes

**Comment by cdarken** [comment]
I found those, I was looking for the actual file too. But I understand now it's an automatic config load. Thanks!

**Comment by root2185** [comment]
> > Yes, by default, it is using that dialogue.
> 
> No, you probably overridden the default, see here -> https://github.com/palinek/nm-tray/blob/master/resources/nm-tray.conf#L2
> 
> It is dead sure, that we can have our own connections editor, but I think it would take a lot of time/work and I'm lacking free time to implement it...

Can you add an option to edit the terminal command to open nmtui-edit?
I don't have xterm here and I'd like to use another.
I made a symbolic link to /usr/bin/tilix to fix it.

**Comment by gfgit** [comment]
Hello everyone. Do you have any news about this?
I'm starting a project about porting LXQt to mobile friendly (discussion [here](https://gitlab.com/postmarketOS/pmaports/-/issues/2406) if you are interested.
This applet is a good candidate for mobile LXQt being it lightweight and with no GTK dependency.

I would like to fill the gaps, like writing a new Qt UI to replace GNOME Network Editor, thus having full Qt experience.
My main goal is to develop a touch friendly interface but I think desktop and mobile UI can live in same executable and maybe switched with a command line argument.

I will do it in my free time so do not expect anything soon, but if you want to collaborate at the code or just send suggestions it's appreciated!

**Comment by palinek** [comment]
> I would like to fill the gaps, like writing a new Qt UI to replace GNOME Network Editor,

It will be very welcome.

**Comment by marklove5102** [comment]
xfce4-terminal  
Works fine after replacement
